### PR TITLE
Allow user to quit GUI from terminal

### PR DESCRIPTION
**Pull request by jarolrod**
Based on #95

This allows a user to quit the GUI from the terminal. It introduces a QTimer in the NodeModel QObject to periodically call the object's `detectShutdown` function.

[Windows](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/97)
[macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/97)
[Android](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/97)

### CODE REVIEW

**Comment by promag** [comment]
I have a preference for exposing the shutdown request to QML, so this looks more promising than #96. From there we can prompt the user or trigger something else before calling `startNodeShutdown`.

I'm fine with both approaches as we can always improve this later on, but we need to fix the quit from terminal.

File: src/qml/nodemodel.h [hunk 1]
```diff
@@ -45,6 +47,7 @@ public Q_SLOTS:
 
     interfaces::Node& m_node;
     std::unique_ptr<interfaces::Handler> m_handler_notify_block_tip;
+    QTimer* pollShutdownTimer;
```
Comment (hebasto) [inline src/qml/nodemodel.h:50]:
```suggestion
    QTimer* m_poll_shutdown_timer{nullptr};
```

**Comment by jarolrod** [comment]
updated from 0c5adc9 -> 0319628

Changes:
- rebase on master
- address hebasto's comments: [m_poll_shutdown_timer](https://github.com/bitcoin-core/gui-qml/pull/97#discussion_r779115197), [directly emit requestedShutdown()](https://github.com/bitcoin-core/gui-qml/pull/95#discussion_r779110627)

File: src/qml/nodemodel.cpp [hunk 2]
```diff
@@ -9,9 +9,14 @@
 
 #include <cassert>
 
+#include <QTimer>
+
 NodeModel::NodeModel(interfaces::Node& node)
     : m_node{node}
 {
+    m_poll_shutdown_timer = new QTimer(this);
```
Comment (promag) [inline src/qml/nodemodel.cpp:17]:
```cpp
auto poll_shutdown_timer = new QTimer(this);
```
and drop the class member.

File: src/qml/nodemodel.cpp [hunk 3]
```diff
@@ -9,9 +9,14 @@
...
+    connect(m_poll_shutdown_timer, &QTimer::timeout, this, &NodeModel::detectShutdown);
```
Comment (promag) [inline src/qml/nodemodel.cpp:18]: Use lamba instead?
```cpp
connect(pool_shutdown_timer, &QTimer::timeout, this, [&] {
    if (m_node.shutdownRequested()) {
        Q_EMIT requestedShutdown();
    }
});
```

**Comment by hebasto** [comment]
Mind looking into an alternative #108?

**Comment by jarolrod** [comment]
Closing in favor of #108

**Pull request closed without merging**